runtime/context.go

```diff
@@ -121,6 +141,11 @@ func annotateContext(ctx context.Context, mux *ServeMux, req *http.Request, rpcM
 				pairs = append(pairs, "authorization", val)
 			}
 			if h, ok := mux.incomingHeaderMatcher(key); ok {
+				if !isValidGRPCMetadataKey(h) {
+					glog.Errorf("HTTP header name %q is not valid as gRPC metadata key; skipping", h)
+					continue
+				}
+
```

Comment:
Nit: I think it's more consistent with our style not to have spurious newlines like this - please remove the ones you added.

Reply:
Like this?